private.tf

```diff
@@ -1,11 +1,11 @@
 module "private_label" {
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
   namespace  = "${var.namespace}"
-  name       = "${var.availability_zone}"
```

Comment:
How to determine in which AZ was created subnets? For example I'm using same names  of subnets in different AZs.

Reply:
We need to maintain a consistent interface in all of our modules. Without supporting the name parameter, we cannot support nesting of the module which would lead to multiple instances of the module.

Reply:
We need to maintain a consistent interface in all of our modules. Without supporting the name parameter, we cannot support nesting of the module which would lead to multiple instances of the module.

Reply:
Ideally, we should use elements from the `names` list here in the `name` field.
But then we need to update `terraform-null-label` with list capability (which whould be pretty easy, but it seems to me out of the scope now).

I thought about the following naming convention:
`<namespace>-<name>-<AZ>-<type>`

But we can't use `<name>` in the middle.
That's why I've updated naming strategy for this module:
`<namespace>-<AZ>-<type>-<name>`

First part can be created by the `label` module using the following assignment:
```
`<namespace>-<name>-<attributes>`
`<namespace>-<AZ>-<type>`
```

Then `<name>` part can be added like this
https://github.com/cloudposse/terraform-aws-named-subnets/pull/4/files#diff-67814bdfab2d6469a77bb872064fde02R18

Only `aws_subnet` uses `label` module, so we don't need a classic `name` argument.

Reply:
@erik , I agree with https://github.com/cloudposse/terraform-aws-named-subnets/pull/4#discussion_r147704458
But then we would have `name` and `names` variables simultaneously.